types/money.yaml

```diff
@@ -0,0 +1,515 @@
+      # TURKS AND CAICOS ISLANDS (THE)
+      # UNITED STATES MINOR OUTLYING ISLANDS (THE)
+      # UNITED STATES OF AMERICA (THE)
+      # VIRGIN ISLANDS (BRITISH)
+      # VIRGIN ISLANDS (U.S.)
+      - USD
+      # 997, US Dollar (Next day), UNITED STATES OF AMERICA (THE)
+      - USN
+      # 940, Uruguay Peso en Unidades Indexadas (UI), URUGUAY
+      - UYI
+      # 858, Peso Uruguayo, URUGUAY
+      - UYU
+      # 927, Unidad Previsional, URUGUAY
+      - UYW
+      # 860, Uzbekistan Sum, UZBEKISTAN
+      - UZS
+      # 926, Bolívar Soberano, VENEZUELA (BOLIVARIAN REPUBLIC OF)
+      - VED
+      # 928, Bolívar Soberano, VENEZUELA (BOLIVARIAN REPUBLIC OF)
+      - VES
+      # 704, Dong, VIET NAM
+      - VND
+      # 548, Vatu, VANUATU
+      - VUV
+      # 882, Tala, SAMOA
+      - WST
+      # 950, CFA Franc BEAC, CAMEROON
+      # CENTRAL AFRICAN REPUBLIC (THE)
+      # CHAD
+      # CONGO (THE)
+      # EQUATORIAL GUINEA
+      # GABON
+      - XAF
+      # 951, East Caribbean Dollar, ANGUILLA
+      # ANTIGUA AND BARBUDA
+      # DOMINICA
+      # GRENADA
+      # MONTSERRAT
+      # SAINT KITTS AND NEVIS
+      # SAINT LUCIA
+      # SAINT VINCENT AND THE GRENADINES
+      - XCD
+      # 960, SDR (Special Drawing Right), INTERNATIONAL MONETARY FUND (IMF) 
+      - XDR
+      # 952, CFA Franc BCEAO, BENIN
+      # BURKINA FASO
+      # CÔTE D'IVOIRE
+      # GUINEA-BISSAU
+      # MALI
+      # NIGER (THE)
+      # SENEGAL
+      # TOGO
+      - XOF
+      # 953, CFP Franc, FRENCH POLYNESIA
+      # NEW CALEDONIA
+      # WALLIS AND FUTUNA
+      - XPF
+      # 994, Sucre, SISTEMA UNITARIO DE COMPENSACION REGIONAL DE PAGOS SUCRE
+      - XSU
+      # 965, ADB Unit of Account, MEMBER COUNTRIES OF THE AFRICAN DEVELOPMENT BANK GROUP
+      - XUA
+      # 886, Yemeni Rial, YEMEN
+      - YER
+      # 710, Rand, LESOTHO
+      - ZAR
+      # 710, Rand, NAMIBIA
+      - ZAR
+      # 710, Rand, SOUTH AFRICA
+      - ZAR
+      # 967, Zambian Kwacha, ZAMBIA
+      - ZMW
+      # 932, Zimbabwe Dollar, ZIMBABWE
+      - ZWL
+  Currency:
+    description: >-
+      Currency object containing the currency code and the accuracy of the minor
+      unit according to ISO 4217.
+    type: object
+    additionalProperties: false
+    required:
+      - code
+    properties:
+      code:
+        description: >-
+          Currency code.
+        type: string
+        $ref: /money#/CurrencyCode
+      decimals:
+        description: >-
+          The number of digits after the decimal separator, that are expressed
+          by the minor units of the currency.
+          For example for Euro, Euro cents express 2 digits after the
+          decimal separator.
+          In other words, the price in the main currency units would be the
+          `value * 10^(-decimals)`.
+        type: integer
+        minimum: 0
+        default: 2
+  Value:
+    description: >-
+      Price is expressed as an amount of minor units of the currency.
```

Comment:
Cost value is expressed as...

Reply:
The type should be named `CostValue`. Or `MoneyAmount` or `MoneyValue`, as we [concluded in conversation above](https://github.com/EVerest/everest-core/pull/425#discussion_r1425308295). Also the description should be adapted, as agardiol pointed out.

Or is the `money` from the filename automatically used as namespace? :thinking: But a namespace isn't available in all programming-languages.